Feature/wc2 jrap

nicholasesposito opened the pull request:
This PR updates the JRAP codes to work on WCOSS2.

nicholasesposito committed:
Updated JRAP. All work except PREP

nicholasesposito committed:
exdump_post.sh edits

nicholasesposito committed:
test

nicholasesposito committed:
get rid of test file "new"

nicholasesposito committed:
Added triggers and update exrap_dump.sh

nicholasesposito committed:
updated pbs scripts

nicholasesposito committed:
Update from bufr_util to bufr_dump

nicholasesposito committed:
Update trigger for ${gfs_ver}

nicholasesposito committed:
Final changes

ilianagenkova commented on jobs/JOBSPROC_RAP_DUMP:
```diff
@@ -36,9 +36,14 @@ echo
+echo "BUFR_DUMP version used by this job, for dump and dumplist, is \
+${bufr_dump_ver:-not set}"
 echo "grib_util module version used by this job is ${grib_util_ver:-not set}"
+if [[ $(echo $LOADEDMODULES | egrep -c "(^|:)bufr_dump/") -eq 0 ]];
```
@ShelleyMelchior-NOAA Nick's check is nicer than my:
echo HOMEbufr_dump is ${HOMEbufr_dump:?"###FATAL ERROR \"HOMEbufr_dump\" is not set"}
Shall I adopt it?

ShelleyMelchior-NOAA replied:
@ilianagenkova Yes!

ilianagenkova commented on jobs/JOBSPROC_RAP_DUMP:
```diff
@@ -143,7 +145,7 @@ export SENDDBN=${SENDDBN:-YES}
 # -------------------------------
 export HOMEobsproc=${HOMEobsproc:\
--${NWROOT:?}/obsproc.${obsproc_ver:?\
+-${PACKAGEROOT:?}/obsproc.${obsproc_ver:?\
```
@nicholasesposito @ShelleyMelchior-NOAA , I replaced "obsproc" in all paths with ${obsNET}. Was that a good move or not?
Nick, in line 148, you need "/" instead of "." between obsproc and obsproc_ver, I think.

ShelleyMelchior-NOAA replied:
@nicholasesposito , @ilianagenkova 's comment is correct. I suggest this:
export HOMEobsproc=${HOMEobsproc:\
-${PACKAGEROOT:?}/${obsNET}/${obsproc_ver:?\

ilianagenkova commented on jobs/JOBSPROC_RAP_PREP:
```diff
@@ -115,7 +120,7 @@ export SENDDBN=${SENDDBN:-YES}
 # -------------------------------
 export HOMEobsproc=${HOMEobsproc:\
--${NWROOT:?}/obsproc.${obsproc_ver:?\
+-${PACKAGEROOT:?}/obsproc.${obsproc_ver:?\
```
@nicholasesposito Replace "." with "/" before the obsproc_ver

ShelleyMelchior-NOAA replied:
See same as JOBSPROC_RAP_DUMP*. Please change to:
export HOMEobsproc=${HOMEobsproc:\
-${PACKAGEROOT:?}/${obsNET}/${obsproc_ver:?\

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP:
```diff
@@ -152,51 +154,51 @@ export SCRIPTSobsproc=${SCRIPTSobsproc:\
-export HOMEbufr_util=${HOMEbufr_util:\
--${NWROOT:?}/bufr_util.${bufr_util_ver:?\
-"###FATAL ERROR \"bufr_util_ver\" is not set"}}
+export HOMEbufr_dump=${HOMEbufr_dump:\
+-${PACKAGEROOT:?}/bufr_dump.${bufr_dump_ver:?\
+"###FATAL ERROR \"bufr_dump_ver\" is not set"}}
```
These 3 lines are redundant and not required. $HOMEbufr_dump is already predefined when the bufr_dump/1.0.0 module was loaded. I suggest removing these lines.

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP:
```diff
@@ -152,51 +154,51 @@ export SCRIPTSobsproc=${SCRIPTSobsproc:\
 # TANK is root directory path to observational database used in dumpjb
 # --------------------------------------------------------------------
-export DCOMROOT=${DCOMROOT:-/gpfs/dell1/nco/ops/dcom}
+export DCOMROOT=${DCOMROOT:-/lfs/h1/ops/canned/dcom}
```
You do not want to point to canned as the default for operations. Please change to:
export DCOMROOT=${DCOMROOT:-/lfs/h1/ops/prod/dcom}

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP:
```diff
@@ -152,51 +154,51 @@ export SCRIPTSobsproc=${SCRIPTSobsproc:\
 # --------------------------------------------------------------------
-export DCOMROOT=${DCOMROOT:-/gpfs/dell1/nco/ops/dcom}
+export DCOMROOT=${DCOMROOT:-/lfs/h1/ops/canned/dcom}
 export TANK=${TANK:-${DCOMROOT}/prod}
```
Note, prod is now included in the $DCOMROOT definition. Please change to:
export TANK=${TANK:-${DCOMROOT}}

ilianagenkova reviewed (commented):
I pointed a couple of things I noticed, but I didn't go over all changes.
If Nick is correct about using ${envir} as the first bit/part passed to compath.py, then I need to correct some of my COM paths... @ShelleyMelchior-NOAA , please tag me when you answer this. thanks!

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP_POST:
```diff
@@ -132,7 +134,7 @@ export SENDDBN=${SENDDBN:-YES}
 # -------------------------------
 export HOMEobsproc=${HOMEobsproc:\
--${NWROOT:?}/obsproc.${obsproc_ver:?\
+-${PACKAGEROOT:?}/obsproc.${obsproc_ver:?\
```
See same as JOBSPROC_RAP_DUMP. Also not missing closed "}". I added it in my example.
export HOMEobsproc=${HOMEobsproc:\
-${PACKAGEROOT:?}/${obsNET}/${obsproc_ver}:?\

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP_POST:
```diff
@@ -142,11 +144,11 @@ export USHobsproc=${USHobsproc:-$HOMEobsproc/ush}
-export HOMEbufr_util=${HOMEbufr_util:\
--${NWROOT:?}/bufr_util.${bufr_util_ver:?\
-"###FATAL ERROR \"bufr_util_ver\" is not set"}}
+export HOMEbufr_dump=${HOMEbufr_dump:\
+-${PACKAGEROOT:?}/bufr_dump.${bufr_dump_ver:?\
+"###FATAL ERROR \"bufr_dump_ver\" is not set"}}
```
See same as JOBSPROC_RAP_DUMP. These three lines are redundant and not required. $HOMEbufr_dump is already predefined when bufr_dump/1.0.0  module is loaded. I suggest you remove these lines.

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_DUMP_POST:
```diff
@@ -170,24 +172,24 @@ cp $UTILROOT/ush/setpdy.sh .; chmod +x setpdy.sh
-export comin=${comin:-${COMIN_ROOT:-${COMROOT:?}}/obsproc/${RUN}.${PDY}}
-export comin_m1=${comin_m1:-${COMIN_ROOT:-${COMROOT:?}}/obsproc/${RUN}.${PDYm1}}
+export comin=${comin:-$(compath.py ${envir}/${obsNET}/${obsproc_ver}/${RUN}.${PDY})}
+export comin_m1=${comin_m1:-$(compath.py ${envir}/${obsNET}/${obsproc_ver}/${RUN}.${PDY})}
```
Be careful ... comin_m1 needs to use PDYm1, not PDY. Change to this:
export comin_m1=${comin_m1:-$(compath.py ${envir}/${obsNET}/${obsproc_ver}/${RUN}.${PDYm1})}

ShelleyMelchior-NOAA commented on jobs/JOBSPROC_RAP_PREP_POST:
```diff
@@ -115,7 +120,7 @@ export SENDDBN=${SENDDBN:-YES}
 # -------------------------------
 export HOMEobsproc=${HOMEobsproc:\
--${NWROOT:?}/obsproc_rap.${obsproc_ver:?\
+-${PACKAGEROOT:?}/obsproc_rap.${obsproc_ver:?\
```
See same as JOBSPROC_RAP_*; also note the missing close "}". change to:
-${PACKAGEROOT:?}/${obsNET}/${obsproc_ver}:?\

ilianagenkova commented:
@Shelley Melchior - NOAA Affiliate ***@***.***> Done!
Updated&pushed my branch.

On Wed, Dec 15, 2021 at 9:17 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In jobs/JOBSPROC_RAP_DUMP
> <https://github.com/NOAA-EMC/obsproc/pull/12#discussion_r770194805>:
>
> >  echo "grib_util module version used by this job is ${grib_util_ver:-not set}"
> +if [[ $(echo $LOADEDMODULES | egrep -c "(^|:)bufr_dump/") -eq 0 ]];
>
> @ilianagenkova <https://github.com/ilianagenkova> Yes!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/obsproc/pull/12#discussion_r770194805>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOC4YXUALTV4A4X3W64TVELURFK2NANCNFSM5KE5MFGA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -361,7 +362,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 1bamua \
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 1bamua \
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained.  Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 1bamua \

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -496,7 +497,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_2} 1 vadwnd \
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_2} 1 vadwnd \
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 vadwnd \

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_2} 1 vadwnd \

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -585,7 +586,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_3} 1 proflr \
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_3} 1 proflr \
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 proflr \

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_3} 1 proflr \

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -675,7 +676,7 @@ fi
 
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_4} 1 msonet gpsipw
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_4} 1 msonet gpsipw
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 msonet gpsipw

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_4} 1 msonet gpsipw

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -761,7 +762,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_5} 1 aircft \
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_5} 1 aircft \
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 aircft \

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_5} 1 aircft \

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -986,7 +987,7 @@ elif [ $cyc -eq 23 ]; then   # (22.50 - 23.49 Z)
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_6} 1 nexrad
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_6} 1 nexrad
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 nexrad

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_6} 1 nexrad

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -1106,7 +1107,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_7} 1 1bhrs4 \
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_7} 1 1bhrs4 \
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 1bhrs4 \

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_7} 1 1bhrs4 \

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -1177,7 +1178,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_8} 1 gsrasr gsrcsr 
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_8} 1 gsrasr gsrcsr 
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 gsrasr gsrcsr

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_8} 1 gsrasr gsrcsr

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -1232,7 +1233,7 @@ else
 fi
 
-$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_9} 1 lghtng
+$USHbufr_dump/bufr_dump_obs.sh $dumptime ${def_time_window_9} 1 lghtng
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_1} 1 lghtng

ShelleyMelchior-NOAA replied:
revised:
$ushscript_dump/bufr_dump_obs.sh $dumptime ${def_time_window_9} 1 lghtng

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -1254,49 +1255,8 @@ set -x
-fi
-if [ "$launcher" = cfp -o "$launcher" = aprun_cfp ]; then
+launcher=${launcher:-"cfp"}
+if [ "$launcher" = cfp ]; then #-o "$launcher" = aprun_cfp ]; then
```
I'd nix everything to the right of the "then". Those are WCOSS1 Cray settings -- not relevant here, and just confusing to be there, even commented.
#-o "$launcher" = aprun_cfp ]; then

ShelleyMelchior-NOAA commented on scripts/exrap_dump.sh:
```diff
@@ -1399,7 +1318,7 @@ set -x
 export STATUS=YES
 export DUMP_NUMBER=10
-$ushscript_dump/bufr_dump_obs.sh $dumptime 3.00 1 null
+$USHbufr_dump/bufr_dump_obs.sh $dumptime 3.00 1 null
```
This change is not required. The original $uscript_dump is defined in JOBSPROC_RAP_DUMP and can be retained. Please revert back to original code:
$ushscript_dump/bufr_dump_obs.sh $dumptime 3.00 1 null

ShelleyMelchior-NOAA commented on trigger_scripts/jrap_prep.wc2.pbs:
```diff
@@ -0,0 +1,144 @@
+export jlogfile=/lfs/h2/emc/stmp/$USER/${JTYP}_prep.${PDY}.jlogfile
+
+export COMIN_ROOT=/lfs/h2/emc/ptmp/$USER/com
+export tstsp=/lfs/h2/emc/ptmp/$USER/com/obsproc/${JTYP}.${PDY}/${JTYP}.t${cyc}z.
```
I suggest commenting out this line or removing it altogether. NCO will never use tstsp. This is more for dev checkout.

ShelleyMelchior-NOAA commented on trigger_scripts/jrap_prep.wc2.pbs:
```diff
@@ -0,0 +1,144 @@
+export COMOUT_ROOT=/lfs/h2/emc/ptmp/$USER/com/
+
+export KEEPDATA=YES
+export POE=NO
+export BACK=NO
+export NSPLIT=1
```
Let's remove or comment these lines for the NCO trigger. These were required for WCOSS Cray where obsproc_rap is currently running in operations. These settings were required in order to force rap_prep to run in serial after the upgrade of GFS to v16. This may not be a requirement for WCOSS2 (*fingers crossed*).

ShelleyMelchior-NOAA reviewed (changes requested):
@nicholasesposito please work through my comments/notes. Thanks!

ShelleyMelchior-NOAA commented:
> 

@ilianagenkova , Nick's usage of compath.py is correct; ${envir} is at the start of the string that is passed into compath.py

nicholasesposito committed:
lots of changes to jobs, triggers, scripts, and CmakeLists.txt

ShelleyMelchior-NOAA committed:
Merge branch 'feature/obsproc_prerelease' into feature/wc2_jrap

ShelleyMelchior-NOAA committed:
Update JOBSPROC_RAP_DUMP

Updated to add comment for clarity where dump and dumplist variables come from
Updated TANK* variable definitions to remove prod

ShelleyMelchior-NOAA committed:
Update JOBSPROC_RAP_DUMP_POST

Added a comment for clarity.
Corrected typo in variable name HOMEarch
Simplified AVGDarch_OUT definition for RUN_ENVIR=nco

ShelleyMelchior-NOAA committed:
Update jrap_prep.wc2.pbs

Commented out settings that are specific for running serially

ShelleyMelchior-NOAA reviewed (approved):
All seems good. I went ahead and made the changes myself and committed them, just to speed up the process.

ShelleyMelchior-NOAA commented:
OK, feature/wc2_jrap is merged into feature/obsproc_prerelease
